test/unit/api/java/SolverTest.java

```diff
@@ -1864,8 +1864,9 @@ void getDifficulty3() throws CVC5ApiException
     // difficulty should map assertions to integer values
     for (Map.Entry<Term, Term> t : dmap.entrySet())
     {
+      // FIXME currently fails:
       assertTrue(t.getKey() == f0 || t.getKey() == f1);
```

Comment:
```suggestion
      assertTrue(t.getKey().equals(f0) || t.getKey().equals(f1));
```
This is the reason of the failure. I checked previous commits and it was me who wrote this and copied it from cpp (I am the one to blame for this). Comparing the equality of terms (and other java objects) should always use `equals` function, whereas `==` operator should only be used when we check reference equality. I don't know why this was passing before, may be the same references were returned in this line, but that change after your PR.

Reply:
Excellent, thanks a lot for looking into this.

It was passing before because this for loop was not being run in the first place, after this PR, it is.